lib/data/datasources/position_source.dart

```diff
@@ -8,23 +9,49 @@ abstract class PositionSource {
   Future<Position> getPosition();
 
   /// Emits the device location in a stream until cancelled.
-  Stream<Position> getPositionStream();
+  ///
+  /// Provide [notificationConfig] to specify the title and text of the
+  /// notification allowing for background location access.
+  Stream<Position> getPositionStream(
+    ForegroundNotificationConfig notificationConfig,
+  );
+
+  Future<bool> isLocationServiceEnabled();
+
+  Future<void> requestLocationService();
 }
 
 @Injectable(as: PositionSource)
 class PositionSourceImpl extends PositionSource {
+  static const _channel = MethodChannel('dev.movna.app/location');
+  static const _enableLocationServiceMethod = 'enable_service';
+
   @override
   Future<Position> getPosition() async {
     return await Geolocator.getCurrentPosition();
   }
 
   @override
-  Stream<Position> getPositionStream() {
+  Stream<Position> getPositionStream(
+    ForegroundNotificationConfig notificationConfig,
+  ) {
     return Geolocator.getPositionStream(
-      locationSettings: const LocationSettings(
+      locationSettings: AndroidSettings(
         // Set to best available accuracy.
         accuracy: LocationAccuracy.best,
+        intervalDuration: const Duration(milliseconds: 500),
+        foregroundNotificationConfig: notificationConfig,
       ),
     );
   }
+
+  @override
+  Future<bool> isLocationServiceEnabled() {
```

Comment:
What is the difference between `LocationService` there and `locationPermission` in the Permission source ?

Reply:
location permission is whether or not the app is authorized to access the device location
`LocationService` is the system service that manages the user's location, this service can be enabled or disabled by the user in the top system bar